Challenges/challenge_46_can_place_flowers/greghilston/src/main/java/Solution.java

```diff
@@ -0,0 +1,54 @@
+public class Solution  {
+    /**
+     * Determines if we can plant a flower at a given location in a specific flowerbed.
+     * @param flowerBed state of flowerbed
+     * @param i location to check
+     * @return whether we can plant a flower or not
+     */
+    private boolean canPlantHere(int[] flowerBed, int i) {
+        // left check
+        int leftNeighborIndex = i - 1;
+
+        if(leftNeighborIndex > 0) // ensures we don't go off the flower bed
+        {
+            if(flowerBed[leftNeighborIndex] == 1) {
```

Comment:
Accessing an int from an array really doesn't cost much, but depending on the context, it might be nice to reduce read access if possible. I also subjectively dislike reading the same data multiple times in a row.
Currently, unless we meet an early return condition, all array fields are read 2-3 times.

Reply:
Understood. Yea I'll probably just punt on this suggestion, but I understand what you're saying. Perhaps I'll take this suggestion if I end up following your first comment above, after I understand that more.